tests/test.rs

```diff
@@ -783,6 +771,45 @@ fn test_parse_number_errors() {
             "number out of range at line 1 column 303",
         ),
     ]);
+
+    // the error messages here vary by implementation
+    test_parse_err::<f64>(&[
+        (
+            "1.a",
+            #[cfg(not(feature = "perfect_float"))]
+            "invalid number at line 1 column 3",
+            #[cfg(feature = "perfect_float")]
+            "trailing characters at line 1 column 3",
+        ),
+        (
+            "100e777777777777777777777777777",
+            #[cfg(not(feature = "perfect_float"))]
+            "number out of range at line 1 column 14",
+            #[cfg(feature = "perfect_float")]
+            "number out of range at line 1 column 31",
+        ),
+        (
+            "-100e777777777777777777777777777",
+            #[cfg(not(feature = "perfect_float"))]
+            "number out of range at line 1 column 15",
+            #[cfg(feature = "perfect_float")]
+            "number out of range at line 1 column 32",
+        ),
+        (
+            "1e+",
+            #[cfg(not(feature = "perfect_float"))]
+            "EOF while parsing a value at line 1 column 3",
+            #[cfg(feature = "perfect_float")]
+            "invalid number at line 1 column 3",
+        ),
+    ]);
+
+    // these parse ok with perfect_float
```

Comment:
This is problematic because they are not valid JSON.

Reply:
This has since been fixed.